packages/phone-number-privacy/src/index.ts

```diff
@@ -1,21 +1,148 @@
+import { AttestationsWrapper } from '@celo/contractkit/lib/wrappers/Attestations'
 import { BLINDBLS } from 'bls12377js-blind'
 import * as functions from 'firebase-functions'
+import fetch from 'node-fetch'
 import config, { connectToDatabase } from './config'
+import { ACCOUNTS_COLUMNS, ACCOUNTS_TABLE } from './models/account'
+import { getContractKit } from './web3/contracts'
+
+export const NUM_ATTESTATIONS_REQUIRED = 3
+
+const contractKit = getContractKit()
+const knex = connectToDatabase()
 
 export const getSalt = functions.https.onRequest((request, response) => {
-  const privateKey = new Buffer(config.salt.key)
+  confirmUser()
+  Promise.resolve(getRemainingQueryCount(request.body.account, request.body.phoneNumber)).then(
+    (remainingQueryCount) => {
+      if (remainingQueryCount <= 0) {
+        response.status(400).send('Requester exceeded salt service query quota')
+        return
+      }
+      const salt = computeBLSSalt(request.body.queryPhoneNumber, response)
+      Promise.resolve(incrementQueryCount(request.body.account))
+        .then(() => {
+          response.json({ success: true, salt })
+          return
+        })
+        .catch((e) => {
+          console.error('Failed to increment query count', e)
+          response.status(500).send('Failed to increment query count')
+          return
+        })
+    },
+    (e) => {
+      console.error('Failed to get query count', e)
+      response.status(500).send('Failed to get query count')
+      return
+    }
+  )
+})
+
+/*
+ * Confirms that user is who they say they are and throws error on failure to confirm.
+ */
+function confirmUser() {
+  //TODO [amyslawson]
+  return
+}
+
+/*
+ * Computes the BLS Salt for the blinded phone number.
+ */
+function computeBLSSalt(queryPhoneNumber: string, response: functions.Response) {
   try {
-    // Adding this here as an example of how to connect to the DB
-    const knex = connectToDatabase()
-    knex('accounts')
-      .first()
-      .then((val) => console.debug('account data', val))
-      .catch((reason) => console.error(reason))
-
-    const salt = BLINDBLS.computePRF(privateKey, new Buffer(request.body.blindPhoneNumber))
-    response.json({ success: true, salt })
+    const privateKey = new Buffer(config.salt.key)
+    return BLINDBLS.computePRF(privateKey, new Buffer(queryPhoneNumber))
   } catch (e) {
-    console.error('Failed to compute BLS salt', e)
-    response.status(500).send('Failed to compute BLS salt')
+    console.error('Failed to compute salt', e)
+    response.status(500).send('Failed to compute salt')
```

Comment:
Don't handle the response directly here. Just let this function worry about BLS and then have the controllers layer (getSalt above) worry about dealing with HTTP codes and such.